### PR TITLE
add transaction lifecycle

### PR DESCRIPTION
**Pull request by TimurParsiq**

### CODE REVIEW

File: src/client/parsiq-client.ts [hunk 1]
```diff
@@ -371,11 +392,14 @@ class ParsiqClient {
...
   private readonly balancesRequestHandler: BalancesRequestHandler;
   private readonly web3HooksRequestHandler: Web3HooksRequestHandler;
+  private readonly transactionLifecycleRequestHandler: TransactionLifecycleRequestHandler;
 
   public setChain(chainId: ChainId) {
     this.tsunamiRequestHandler.setChain(chainId);
     this.nftRequestHandler.setChain(chainId);
     this.balancesRequestHandler.setChain(chainId);
+    this.web3HooksRequestHandler.setChain(chainId);
+    this.web3HooksRequestHandler.setChain(chainId);
```
Comment (daniil-parsiq) [inline src/client/parsiq-client.ts]: two times web3hooks

Comment (TimurParsiq) [reply src/client/parsiq-client.ts]: good catch! changed!

**Comment by daniil-parsiq** [comment]
I'm generally worried about the naming. Looking at this function name - `listTransactionLifecycles`, it seems a bit ridiculous. Transaction Lifecycle is the service's name, but I don't think we should use it as the entity name. Yet I'm struggling with a better naming, maybe just something like `createTransactionLifecycleRequest`, `listTransactionLifecycleRequests`. WDYT?

File: src/client/parsiq-client.ts [hunk 2]
```diff
@@ -59,8 +61,27 @@ class ParsiqClient {
...
+  public readonly txLifecycle = {
+    create: (createTransactionLifecycle: CreateTransactionLifecycle) => {
+      return this.transactionLifecycleRequestHandler.createTransactionLifecycle(createTransactionLifecycle);
+    },
+
+    list: () => {
+      return this.transactionLifecycleRequestHandler.listTransactionLifecycles();
+    },
+
+    show: (id: string) => {
```
Comment (daniil-parsiq) [inline src/client/parsiq-client.ts]: I think in both cases, tx lifecycle and web3hooks the better naming is `get`?

Comment (konstantinkoniukov) [reply src/client/parsiq-client.ts]: createTxLcHook OR createTxLifecycleHook
listTxLcHooks
showTxLcHook
deleteTxLcHook

just to keep it short

Comment (konstantinkoniukov) [reply src/client/parsiq-client.ts]: or let's just shorten Danny's options:

createTxLcRequest, listTxLcRequests...

File: src/client/transaction-lifecycle-hooks-request-handler.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    config: { axiosConfig?: AxiosRequestConfig; retryConfig?: IAxiosRetryConfig } = {
+      axiosConfig: {},
+      retryConfig: {},
+    },
+  ) {
+    const { axiosConfig = {}, retryConfig = {} } = config;
+    super(TSUNAMI_BASE_URL, chain, apiKey, axiosConfig, retryConfig);
+  }
+
+  public async createTxLcHook(createHook: CreateTransactionLifecycle): Promise<string> {
```
Comment (daniil-parsiq) [inline src/client/transaction-lifecycle-hooks-request-handler.ts:22]: why `createTxLcHook` (etc), if it's already in the TransactionLifecycle namespace?

Comment (daniil-parsiq) [reply src/client/transaction-lifecycle-hooks-request-handler.ts:22]: i guess it's ±irrelevant, since in the public interface it's without redundancy

Comment (TimurParsiq) [reply src/client/transaction-lifecycle-hooks-request-handler.ts:22]: not needed strictly, but other handlers do have specifics, so rather to be on par with the rest than for necessity

File: examples/transaction-lifecycles.ts
Comment (konstantinkoniukov) [inline examples/transaction-lifecycles.ts:1]: уберём 's' после lifecycle?
то есть: если мы используем transaction lifecycle, то всегда в единственном.
hooks во множественном имеет смысл для list, а для create, show,  и delete один hook.
по крайней мере, такая логика в доке

File: examples/transaction-lifecycle-hooks.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  const hooks = await client.txLifecycleHooks.list();
+  console.log(`Total of ${hooks.length} transaction lifecycle(s)`)
+  for(const hook of hooks) {
+    console.log(JSON.stringify(hook));
+  }
+
+  await client.txLifecycleHooks.delete(id);
+}
+
+runTransactionLifecycles();
```
Comment (konstantinkoniukov) [inline examples/transaction-lifecycle-hooks.ts:35]: уберём 's' после lifecycle?
то есть: если мы используем transaction lifecycle, то всегда в единственном.
hooks во множественном имеет смысл для list, а для create, show, и delete один hook.
по крайней мере, такая логика в доке

File: src/client/transaction-lifecycle-hooks-request-handler.ts (hunk 3 above)
Comment (daniil-parsiq) [reply src/client/transaction-lifecycle-hooks-request-handler.ts:22]: yeah, sorry - now i see difference between public interface and request handlers

**Review by daniil-parsiq — APPROVED** [review]
lgtm

File: examples/web3-hooks.ts [hunk 5]
```diff
@@ -5,31 +5,31 @@ import { HttpAuthType } from "../src";
...
   console.log(`Total of ${hooks.length} hook(s)`)
   for(const hook of hooks) {
     console.log(JSON.stringify(hook));
   }
 
-  await client.web3hooks.delete(id);
+  await client.web3Hooks.delete(id);
 }
 
 runWeb3Hooks();
```
Comment (konstantinkoniukov) [inline examples/web3-hooks.ts:35]: я вижу, что мы используем web3hookS во многих случаях. в доке в единственном числе, но полагаю это не проблема и можно оставить так.

**Pull request merged**